sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/hints.scala

```diff
@@ -213,6 +213,11 @@ trait HintErrorHandler {
    */
   def joinNotFoundForJoinHint(hint: HintInfo): Unit
 
+  /**
+   * Callback for a join hint specified on a join that doesn't support this build side.
```

Comment:
if we only target for join hint, let's name it `joinHintNotSupported`

Reply:
done